### PR TITLE
Check key should be string in nn.Module.add_module, parameter and buffer

### PR DESCRIPTION
**Pull request by karandwivedi42**
cc @SsnL

### CODE REVIEW

**Review by ssnl — CHANGES_REQUESTED** [review]

**Review by ssnl — APPROVED** [review]
lgtm

File: torch/nn/modules/module.py [hunk 1]
```diff
@@ -101,12 +101,15 @@ def register_buffer(self, name, tensor):
             >>> self.register_buffer('running_mean', torch.zeros(num_features))
 
         """
-        if hasattr(self, name) and name not in self._buffers:
-            raise KeyError("attribute '{}' already exists".format(name))
+        if name is None or not isinstance(name, torch._six.string_classes):
```
Comment (SsnL) [inline torch/nn/modules/module.py]: This `is None` check seems unnecessary. The `isinstance` will catch it anyways. Can you remove it?

**Comment by SsnL** [comment]
Thanks! Can you pull & rebase to upstream master please? We made some changes to the system so current PRs needs to be rebased to master to land. (Wont be a problem for future PRs). Sorry about it!

**Comment by karandwivedi42** [comment]
Reopened in https://github.com/pytorch/pytorch/pull/8960

**Comment by SsnL** [comment]
lol I just went to your branch and fixed the history. I guess that also works :)

**Pull request closed without merging**